### PR TITLE
asset: Only persistent data to disk for targetable assets and tests refactor

### PR DESCRIPTION
**Pull request by yifan-gu**
- Only assets that will be fetched by the command line will be written to the disk.
- Use t.Run() for iterating test cases.
- Use assert package more.

### CODE REVIEW

File: cmd/openshift-install/main.go [hunk 1]
```diff
@@ -37,8 +37,14 @@ func main() {
 	log.SetLevel(l)
 
 	assetStore := &asset.StoreImpl{}
-	if _, err := assetStore.Fetch(targetAsset); err != nil {
+	st, err := assetStore.Fetch(targetAsset)
```
Comment (abhinavdahiya) [inline cmd/openshift-install/main.go]: There can be multiple assets in a single target. For now it is okay.

Comment (yifan-gu) [reply cmd/openshift-install/main.go]: I think the contents of a target can just include all its assets.

**Comment by wking** [comment]
Looks like the Go unit tests [hung][1]:

```
018/08/30 23:18:36 Building src {"component":"entrypoint","level":"error","msg":"Process did not finish before 4h0m0s timeout","time":"2018-08-31T03:18:34Z"}
```

Maybe we broke something...

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/196/pull-ci-origin-installer-unit/643/build-log.txt

**Pull request merged**